### ISSUE TITLE
Stream does not open + no audio sources listed on menu

### ISSUE THREAD

**Issue by Grandchild1** [maltejur/discord-screenaudio#39]
Hi I am using Ubuntu and the app opens, works well, but as soon as I try to start streaming it shows the list of things I can pass through but only desktop audio is shown despite me playing another audio source in the background. When I select desktop audio the menu closes, no stream gets opened, and when I press the screen share option again, nothing opens and nothing happens.
  I am a little bit new to Linux, only used it for a few months now so I don't know technical stuff but I do have a good read on my way around the OS and use of the terminal.

Thank you,
Grandchild

**Comment by Blackcatmaxy** [comment]
Exactly tame issue with Flatpak on OpenSUSE KDE (also tried package, but that didn't make a .desktop file?), looking at the screenshots on #30 and on the readme I'm not sure if there's some Discord update that broke this since it looks identical to normal Discord and doesn't have the third button next to the camera/screenshare buttons. This happened both on my first try in X11 and in Wayland.

**Comment by maltejur** [comment]
Hm, I can not reproduce this on my laptop. I am back home again in a few days, I'll try again then in a VM.

Are you both on PipeWire and have the language set to English?

**Comment by Grandchild1** [comment]
I am glad I am not the only one with this issue. I indeed do use PipeWire and I have my language set to English. What is bizarre is that it works on Mint Cinnamon by installing PipeWire, and it works on Pop_OS!. Which doesn't make too much sense to me because they are both based on Ubuntu. Thank you for looking into it soon.

**Comment by Grandchild1** [comment]
> Exactly same issue with Flatpak on OpenSUSE KDE (also tried package, but that didn't make a .desktop file?), looking at the screenshots on #30 and on the readme I'm not sure if there's some Discord update that broke this since it looks identical to normal Discord and doesn't have the third button next to the camera/screenshare buttons. This happened both on my first try in X11 and in Wayland.
> 
> EDIT for more details, like previous report said, clicking screenshare button asks about which audio source to share and screensize/framerate details (with only options being full desktop audio or none) and then when after that nothing open up to ask what screen to use. And the screenshare button won't work until you rejoin the VC.

Perhaps it could be an issue with out of the box Flatpak support on distros, but that would also be weird because after installing Flatpak and everything else to support it, my other Flatpak apps work completely fine so it is pretty strange.

**Comment by maltejur** [comment]
Did you also manually install PipeWire on Ubuntu

> Perhaps it could be an issue with out of the box Flatpak support on distros

I doubt it, the Flatpak works just fine on Arch, where you also have to install it manually

**Comment by Grandchild1** [comment]
It is there by default on Ubuntu, yeah

**Comment by Blackcatmaxy** [comment]
> Are you both on PipeWire and have the language set to English?

yep have pipewire and English as my only language (although I do have spanish spell check enabled but that shouldn't be related?)

**Comment by maltejur** [comment]
Ok, can you maybe post the program output when the problem happens?

**Comment by Blackcatmaxy** [comment]
Seems like it's flooding with `[userscript] Did not find 'discord-screenaudio-virtmic', trying again in 100ms`
> [discord] "BaseWebRTCConnection.updateVideoQuality: resolutionScaleFactor: 1, encodingVideoWidth: undefined, remoteSinkWantsMaxFramerate: 30, encodingVideoMaxBitRate: 2500000, localWant: 100"
[discord] "%c[UnifiedConnection] \nfont-weight: bold;\ncolor: purple;\n signalingState => stable"
[discord] "updateVideoQuality: {\n    \"remoteSinkWantsMaxFramerate\": 30\n}"
[discord] "BaseWebRTCConnection.updateVideoQuality: resolutionScaleFactor: undefined, encodingVideoWidth: undefined, remoteSinkWantsMaxFramerate: 30, encodingVideoMaxBitRate: undefined, localWant: 100"
[virtmic] Stopping Virtmic
[virtmic] Starting Virtmic with target "[All Desktop Audio]"
[userscript] Did not find 'discord-screenaudio-virtmic', trying again in 100ms
[discord] "%c[RTCControlSocket] \nfont-weight: bold;\ncolor: purple;\n Sending heartbeat"
[discord] "%c[RTCControlSocket] \nfont-weight: bold;\ncolor: purple;\n Heartbeat ACK received"
[userscript] Did not find 'discord-screenaudio-virtmic', trying again in 100ms
[userscript] Did not find 'discord-screenaudio-virtmic', trying again in 100ms
[userscript] Did not find 'discord-screenaudio-virtmic', trying again in 100ms

**Comment by maltejur** [comment]
Does it stop after a few seconds or does it just keep on going?

**Comment by Blackcatmaxy** [comment]
Keep going

**Comment by extract** [comment]
I'm seeing same issue both using flatpak and through AUR. Same spam as Blackcatmaxy mentioned.

**Comment by Grandchild1** [comment]
> Seems like it's flooding with `[userscript] Did not find 'discord-screenaudio-virtmic', trying again in 100ms` (small snippet, can put a hastebin of full output if needed)
> 
> > [discord] "BaseWebRTCConnection.updateVideoQuality: resolutionScaleFactor: 1, encodingVideoWidth: undefined, remoteSinkWantsMaxFramerate: 30, encodingVideoMaxBitRate: 2500000, localWant: 100"
> > [discord] "%c[UnifiedConnection] \nfont-weight: bold;\ncolor: purple;\n signalingState => stable"
> > [discord] "updateVideoQuality: {\n    "remoteSinkWantsMaxFramerate": 30\n}"
> > [discord] "BaseWebRTCConnection.updateVideoQuality: resolutionScaleFactor: undefined, encodingVideoWidth: undefined, remoteSinkWantsMaxFramerate: 30, encodingVideoMaxBitRate: undefined, localWant: 100"
> > [virtmic] Stopping Virtmic
> > [virtmic] Starting Virtmic with target "[All Desktop Audio]"
> > [userscript] Did not find 'discord-screenaudio-virtmic', trying again in 100ms
> > [discord] "%c[RTCControlSocket] \nfont-weight: bold;\ncolor: purple;\n Sending heartbeat"
> > [discord] "%c[RTCControlSocket] \nfont-weight: bold;\ncolor: purple;\n Heartbeat ACK received"
> > [userscript] Did not find 'discord-screenaudio-virtmic', trying again in 100ms
> > [userscript] Did not find 'discord-screenaudio-virtmic', trying again in 100ms
> > [userscript] Did not find 'discord-screenaudio-virtmic', trying again in 100ms

Looking at the microphone selection on discord, this does confirm that it simply flat out does not recognize any of my input or output devices nor does it list the virtual microphone. I suspect that since its missing it is causing all of these issues.

**Comment by Grandchild1** [comment]
sorry didnt mean to do that

**Comment by maltejur** [comment]
Hm, I wasn't able to reproduce this in OpenSUSE Tumbleweed either. Does the normal microphone in discord work or do you also have no audio there?

**Comment by Blackcatmaxy** [comment]
Normal audio works, but unlike desktop discord it only shows "Default" Input and "Default" Output devices and doesn't show any other options in the dropdowns.

**Comment by Blackcatmaxy** [comment]
It also doesn't recognize my camera

**Comment by Blackcatmaxy** [comment]
Could this be related? Decided to put my [whole log output into a hastebin](https://hastebin.com/zatukibali.lua) to see if it could help. There seem to be some errors around like `[default] QLayout: Attempting to add QLayout "" to StreamDialog "", which already has a layout` and `[32376:61:0904/185315.364744:ERROR:audio_rtp_receiver.cc(90)] AudioRtpReceiver::OnSetVolume: No audio channel exists`, as well as several deprecation warnings?

**Comment by Blackcatmaxy** [comment]
Hmm seems like I was wrong and even though I had pipewire installed, the service was not enabled. Checked on my (wayland) laptop and it definitely did have pipewire enabled but still had the same output as well as two more errors 
```
Failed to create wl_display (No such file or directory)
qt.qpa.plugin: Could not load the Qt platform plugin "wayland" in "" even though it was found.
```

**Comment by Blackcatmaxy** [comment]
Huh I never knew pipewire was so complicated, recommend a section in the readme telling people if they have issues to run `pactl info` and if Servername is not "Pulsaudio (on PipeWire x.y.z)" then their set up is not correct. Even on my laptop I was missing `pipewire-pulseaudio` which after now installing and setting up per [the OpenSUSE docs](https://en.opensuse.org/openSUSE:Pipewire) finally lets me share screens with this.
